engine/resource/src/dmsdk/resource/resource.h

```diff
@@ -257,10 +256,41 @@ bool ResourcePreloadHint(HResourcePreloadHintInfo preloader, const char* path);
  * @param factory [type: HResourceFactory] Factory handle
  * @param resource [type: void*] The resource pointer
  * @param hash [type: dmhash_t*] (out) The path hash of the resource
- * @return result [type: ResourceResult] RESULT_OK on success
+ * @return result [type: ResourceResult] RESOURCE_RESULT_OK on success
  */
 ResourceResult ResourceGetPath(HResourceFactory factory, const void* resource, dmhash_t* hash);
 
+/*#
+ * Get a resource extension from a path, i.e "resource.ext" will return ".ext". Note the included dot in the output.
+ * @name ResourceGetExtFromPath
+ * @param path [type: const char*] The path to the resource
+ * @return extension [type: const char*] Pointer to extension string if an extension was found, 0 otherwise
+ */
+const char* ResourceGetExtFromPath(const char* path);
```

Comment:
[Suggestion] Maybe it's worth to return without dot to be consistent with `ResourceGetTypeFromExtension` which expects extension without dot?